Rake task to add calendars to the search index

carvil opened the pull request:
Previously, panopticon would send artifacts to Rummager. That flow is now being
deprecated in favour of a simpler approach which is to send documents directly
to rummager.

This commit adds a class that allows us to add calendars to the search index
directly. It also adds a rake task `rummager:index_all` which goes through all
calendars and indexes them in the search index.

Before, the search results in `search-admin` looked like this:

<img width="1182" alt="screen shot 2016-09-19 at 10 31 16" src="https://cloud.githubusercontent.com/assets/416701/18627785/cf97fd58-7e54-11e6-9a46-62a314176f8d.png">

After the change, it looks like this:

<img width="1199" alt="screen shot 2016-09-19 at 11 21 10" src="https://cloud.githubusercontent.com/assets/416701/18629619/a5fd124e-7e5e-11e6-9c16-2dbc3c86596f.png">

The payload now includes the following extra attributes:
- publishing_app
- rendering_app
- content_id

Trello ticket: https://trello.com/c/gsHnT5WF/161-epic-send-things-to-rummager-directly-instead-of-via-panopticon

tijmenb commented on app/services/search_payload.rb:
```diff
@@ -0,0 +1,36 @@
+  end
+
+  def call
+    {
```
Some of these attributes (`owning_app`, `paths` and more)  are not valid in rummager. Rummager will silently discard them probably, but good to remove anyway.

You can check [field_definitions.json](https://github.com/alphagov/rummager/blob/master/config/schema/field_definitions.json) for all possible attributes, and [document_types/edition.json](https://github.com/alphagov/rummager/blob/master/config/schema/document_types/edition.json) for all field names allowed in an `edition`.

tijmenb commented on lib/services.rb:
```diff
@@ -7,4 +8,8 @@ def self.publishing_api
     )
   end
+
+  def self.rummager_api
```
I think `Services.rummager` instead of `Services.rummager_api` would be more consistent with other applications.

carvil replied:
Will change

mobaig commented on app/services/search_payload.rb:
```diff
@@ -0,0 +1,36 @@
+class SearchPayload
```
Reckon this should be in `app/presenters`.

carvil replied:
👍 will change

mobaig commented on lib/tasks/rummager.rake:
```diff
@@ -0,0 +1,17 @@
+require 'registerable_calendar'
+
+namespace :rummager do
+  desc "Register application metadata with panopticon"
```
desc needs updating.

carvil replied:
Yup, good spot

mobaig commented on app/services/search_payload.rb:
```diff
@@ -0,0 +1,36 @@
+class SearchPayload
+  attr_reader :registerable_calendar
```
We can probably drop the 'registerable' terminology altogether now that we've moved this away from panopticon, referring to this simply as `calendar`.

carvil replied:
I didn't want to change the model name and thought it would be better to be explicit about what this is. There is a Calendar model in the app, but it's incompatible with this class.

mobaig commented on lib/tasks/rummager.rake:
```diff
@@ -0,0 +1,17 @@
+namespace :rummager do
+  desc "Register application metadata with panopticon"
+  task register: :environment do
+    require 'gds_api/rummager'
```
if we're loading the environment we probably don't need this?

carvil commented:
@tijmenb and @mobaig I have addressed your comments and updated the screenshots above, WDYT? Can we use this as a template for the rest of the changes?
